haystack/nodes/prompt/prompt_template.py

```diff
@@ -268,9 +268,14 @@ def prepare(self, *args, **kwargs) -> Dict[str, Any]:
                 if param in kwargs:
                     params_dict[param] = kwargs[param]
 
-        if set(params_dict.keys()) != set(self.prompt_params):
+        if not set(self.prompt_params).issubset(set(params_dict.keys())):
```

Comment:
```suggestion
        if not set(self.prompt_params).issubset(params_dict.keys()):
```

`issubset` can be any iterable, no need to create an unnecessary set.